README.md

```diff
@@ -54,30 +129,60 @@ const credential = await navigator.credentials.get({
+Any origin may invoke the [Payment Request API](https://w3c.github.io/payment-request/) with the `secure-payment-confirmation` payment method to prompt the user to verify a SecurePaymentCredential created by any other origin. The `PaymentRequest.show()` method must require a user gesture. The browser will display a native user interface with the payment amount and the payee origin, which is taken to be the origin of the top-level context where the `PaymentRequest` API was invoked.
+
+Proposed new `secure-payment-confirmation` payment method:
+
+```webidl
+dictionary SecurePaymentConfirmationRequest {
+  SecurePaymentConfirmationAction action;
+  required DOMString instrumentId;
+  required BufferSource networkData;
+  unsigned long timeout;
+  required USVString fallbackUrl;
+};
+
+enum SecurePaymentConfirmationAction {
+  "authenticate",
+};
+
+dictionary SecurePaymentConfirmationResponse {
+  required SecurePaymentConfirmationMerchantData merchantData;
+  required BufferSource networkData;
+  required AuthenticatorAssertionResponse assertion;
+};
+
+dictionary SecurePaymentConfirmationMerchantData {
+  required USVString merchantOrigin;
+  required PaymentCurrencyAmount total;
+};
+```
+
+Example usage:
 
 ```javascript
-const publicKeyCredentialRequestOptions = {
+const securePaymentConfirmationRequest = {
+  action: "authenticate",
+  instrumentId: "Q1J4AwSWD4Dx6q1DTo0MB21XDAV76",
   challenge,
-  allowCredentials: [{
-    id: ['ADSUllKQmbqdGtpu4sjseh4cg2TxSvrbcHDTBsv4NSSX9...'],
-    type,
-    transports,
-  }],
   timeout,
+  fallbackUrl: "https://fallback.example/url"
 };
 
 const request = new PaymentRequest(
   [{supportedMethods: 'secure-payment-confirmation',
-    data: {
-      publicKey: publicKeyCredentialRequestOptions,
-      fallback: 'https://fallback.example/url',
-    },
+    data: securePaymentConfirmationRequest
   }],
   {total: {label: 'total', amount: {currency: 'USD', value: '20.00'}}});
 const response = await request.show();
+
+// Merchant validates |response.merchantData| and sends |response| to issuer for authentication.
 ```
 
-If none of the credentials from `allowCredentials.id` are available, then the user agent will invoke the payment handler for the payment method specified in the `fallback` method. It is recommended that the `fallback` payment method supports just-in-time installation of a payment handler, so the user agent may install it just-in-time. This enables the payment handler to open a fallback URL in the payment handler window.
+If the payment instrument specified by `instrumentId` is not available or if the user failed to authenticate, then the
+user agent will open `fallbackUrl` inside the Secure Modal Window.
```

Comment:
Secure Modal Window is a UI paradigm that does not define any communication channel to the merchant or the browser at this time. Should `PaymentRequest.show()` return some kind of indicator to the merchant that this has happened and the browser will not be showing an authentication prompt to the user?

Reply:
There's an open question for a payment handler should be involved in a "secure-payment-confirmation" transaction. If it is, then the most natural way is probably to expose the error via an extension to `PaymentRequestEvent`. But if a payment handler should not be involved, maybe the browser should build a new communication channel for the web content inside a Secure Modal Window to communicate with the browser. More discussion around this is needed.

For now, I just called out the open question, and left a suggested hack for the pilot.